solc/main.cpp

```diff
@@ -53,24 +52,33 @@ static void setDefaultOrCLocale()
 
 int main(int argc, char** argv)
 {
-	setDefaultOrCLocale();
-	solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
-	if (!cli.parseArguments(argc, argv))
-		return 1;
-	if (!cli.readInputFiles())
+	try
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception || Exception const& _exception || InternalCompilerError const& _exception || smtutil::SMTLogicError const& _exception)
```

Comment:
I meant that this `||` for exceptions shouldn't compile. You would probably have to write different catch blocks.